geniza/corpus/models.py

```diff
@@ -334,6 +333,15 @@ def shelfmark(self):
             )
         )
 
+    @property
+    def shelfmark_display(self):
+        """First shelfmark plus join indicator for shorter display."""
+        certain = self.textblock_set.filter(certain=True)
+        if not certain.exists():
+            return None
```

Comment:
Is this allowed / possible? Shouldn't we expect at least one definite shelfmark if we have a document?

Reply:
It might not be allowed, but it's definitely possible because we can't require that a document have any `TextBlocks`, right? Added because tests will fail on newly created documents since they have no associated fragments at all and it seemed better to handle it.

Reply:
Although maybe we should add a `save()` hook to just prevent that from happening, actually, since it definitely doesn't make sense to have a document without any textblocks